packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/mtom-parser.ts

```diff
@@ -0,0 +1,156 @@
+
+type MtomContentType = {
+  boundary: string;
+  type: string;
+  start: string;
+  startInfo?: string | undefined;
+};
+
+function parseMtomContentType(contentType: string): MtomContentType {
+  const contentTypeParams = contentType.split(";").reduce<Record<string, string>>((acc, param) => {
+    const index = param.indexOf("=");
+    if (index !== -1) {
+      const key = param.substring(0, index).trim().toLowerCase();
+      const value = param
+        .substring(index + 1)
+        .trim()
+        .replace(/"/g, "");
+      acc[key] = value;
+    }
+    return acc;
+  }, {});
+
+  if (!contentTypeParams.boundary) {
+    throw new Error("No boundary parameter found in content type.");
+  }
+  if (!contentTypeParams.type) {
+    throw new Error("No type parameter found in content type.");
+  }
+  if (!contentTypeParams.start) {
+    throw new Error("No start parameter found in content type.");
+  }
+  return {
+    boundary: contentTypeParams.boundary,
+    type: contentTypeParams.type,
+    start: contentTypeParams.start,
+    startInfo: contentTypeParams["start-info"],
+  };
+}
+
+type MtomHeaders = {
+  ContentID: string;
+  ContentTransferEncoding?: string | undefined;
+  ContentType: string;
+};
+
+function parseMtomHeaders(headerPart: string): MtomHeaders {
+  const headers = headerPart.split("\n").reduce<Record<string, string>>((acc, headerLine) => {
+    const index = headerLine.indexOf(":");
+    if (index !== -1) {
+      const key = headerLine.substring(0, index).trim().toLowerCase();
+      const value = headerLine.substring(index + 1).trim();
+      acc[key] = value;
+    }
+    return acc;
+  }, {});
+
+  if (!headers["content-id"]) {
+    throw new Error("No Content-ID header found in headers.");
+  }
+  if (!headers["content-type"]) {
+    throw new Error("No Content-Type header found in headers.");
+  }
+  return {
+    ContentID: stripTags(headers["content-id"]),
+    ContentTransferEncoding: headers["content-transfer-encoding"],
+    ContentType: headers["content-type"],
+  };
+}
+
+export function parseMTOMResponse(mtomMessage: string, contentType: string): DocumentResponse[] {
+  const contentTypeParams = parseMtomContentType(contentType);
+
+  const boundary = `--${contentTypeParams.boundary}`;
+  const parts = mtomMessage.split(boundary).slice(1, -1);
+
+  const documentResponses: DocumentResponse[] = [];
+  const attachments: Record<string, string> = {};
+
+  parts.forEach(part => {
+    let splitter = "\r\n\r\n";
+    let headersEndIndex = -1;
+    if (contentTypeParams.startInfo) {
+      splitter = contentTypeParams.startInfo + splitter;
+      headersEndIndex = part.indexOf(splitter);
+    }
+    if (headersEndIndex === -1) {
+      splitter = "\r\n\r\n";
+      headersEndIndex = part.indexOf(splitter);
+      if (headersEndIndex === -1) {
+        splitter = "\n\n";
+        headersEndIndex = part.indexOf(splitter);
+        if (headersEndIndex === -1) {
```

Comment:
this is where we look for the split of the header. Basically, the first message within the mtom should start with the startInfo specified in the contentType followed by the the carriage Return Line Feed (CRLF). 

Subsequent messages within the same MTOM will just start with the the CRLF, and a few edge cases I saw where messages weren't conforming to the spec just used two new lines.